internal/testing/enginetest/enginetest.go

```diff
@@ -362,19 +365,29 @@ func runTestModuleEngine_Call_HostFn_ModuleContext(t *testing.T, et EngineTester
 
 	t.Run("defaults to module memory when call stack empty", func(t *testing.T) {
 		// When calling a host func directly, there may be no stack. This ensures the module's memory is used.
-		results, err := me.Call(testCtx, modCtx, fns[0], 3)
+		results, err := me.Call(testCtx, modCtx, fns[0])
 		require.NoError(t, err)
 		require.Equal(t, uint64(3), results[0])
-		require.Same(t, memory, mMemory)
 	})
 }
 
 func RunTestModuleEngine_Call_HostFn(t *testing.T, et EngineTester) {
-	runTestModuleEngine_Call_HostFn_ModuleContext(t, et) // TODO: refactor to use the same test interface.
+	t.Run("wasm", func(t *testing.T) {
+		runTestModuleEngine_Call_HostFn(t, et, hostDivByWasm)
+		if !et.IsCompiler() { // TODO: Support host wasm func that uses caller memory in compiler.
```

Comment:
This is the TODO which is likely a lot harder, to have the compiler look at ir.IsHostFunction && ir.UsesMemory to allow wasm to read the memory from the caller not the defining module.